autosklearn/ensemble_builder.py

```diff
@@ -950,13 +929,20 @@ def _delete_excess_models(self):
             for lock in locks:
                 lock.release()
 
-    def _read_np_fn(self, fp):
-        if self.precision == "16":
-            predictions = np.load(fp, allow_pickle=True).astype(dtype=np.float16)
-        elif self.precision == "32":
-            predictions = np.load(fp, allow_pickle=True).astype(dtype=np.float32)
-        elif self.precision == "64":
-            predictions = np.load(fp, allow_pickle=True).astype(dtype=np.float64)
+    def _read_np_fn(self, path):
+        if path.endswith("gz"):
+            open_method = gzip.open
+        elif path.endswith("npy"):
+            open_method = open
         else:
-            predictions = np.load(fp, allow_pickle=True)
-        return predictions
+            raise ValueError("Unknown filetype %s" % path)
+        with open_method(path, 'rb') as fp:
+            if self.precision == 16:
```

Comment:
Hey, I don't know whether the string-based version is still used anywhere. Could you please change the code to check for both strings and integers, but also issue a warning if the precision is a string?